client/navigation/url-hotkey.ts

```diff
@@ -0,0 +1,50 @@
+import { useRoute } from 'wouter'
+import { useKeyListener } from '../keyboard/key-listener'
+import { HotkeyProp } from '../material/button'
+import { push } from './routing'
+
+export interface UrlHotkeyProps {
+  /** URL to navigate to. */
+  url: string
+  /**
+   * A hotkey, or an array of hotkeys, to register for the URL. Pressing any of the specified
+   * modifiers and key combinations will result in the page navigation to the given URL.
+   */
+  hotkey: HotkeyProp | HotkeyProp[]
+  /** Whether the navigation is disabled (hotkey will do nothing). */
+  disabled?: boolean
+  /** A function that will perform the transition to the new URL. */
+  transitionFn?: (url: string) => void
+}
+
+/**
+ * A hook which allows any component to register a hotkey (or multiple hotkeys) which upon pressing
+ * will navigate the application to the given URL.
+ *
+ * If the currently active URL is the same as the given one, pressing the hotkey will do nothing.
+ */
+export function useUrlHotkey({ url, hotkey, disabled, transitionFn = push }: UrlHotkeyProps) {
```

Comment:
This hook seems unnecessary, you could just click the link in the same way that we click any other button, and then it would also still activate the ripple and such?

If you really have to have it (and I would very much that we don't), at least rename it `useNavigationHotkey`